### PR TITLE
Add edit and display of persons on books

### PR DESCRIPTION
**Pull request by Pithaya**
Allow adding persons on books, and show them on the item details page.
Related PR on the server: [#10371](https://github.com/jellyfin/jellyfin/pull/10371)

**Changes**
- Show the "people" section of the metadata editor for book items
- Add new values to the person type select
- Add new translation keys
- Show "book specific" person types (author, editor and translator) below genres, and "comics specific" person types (illustrator, inker, letterer, etc.) as cards
- Update CONTRIBUTORS.md

**Issues**

### CODE REVIEW

File: src/controllers/itemDetails/index.js [hunk 1]
```diff
@@ -1002,6 +1003,93 @@ function renderStudio(page, item, context) {
...
+    authorsLabel.innerHTML = globalize.translate(authors.length > 1 ? 'Authors' : 'Author');
+    const authorsValue = page.querySelector('.authors');
+    authorsValue.innerHTML = html;
+
+    const authorsGroup = page.querySelector('.authorsGroup');
+    if (authors.length) {
+        authorsGroup.classList.remove('hide');
+    } else {
+        authorsGroup.classList.add('hide');
+    }
```
Comment (dmitrylyzo) [inline src/controllers/itemDetails/index.js:1032]: You can use `toggle`.
```suggestion
    authorsGroup.classList.toggle('hide', !authors.length);
```

File: src/controllers/itemDetails/index.js [hunk 2]
```diff
@@ -1002,6 +1003,93 @@ function renderStudio(page, item, context) {
...
+    editorsLabel.innerHTML = globalize.translate(editors.length > 1 ? 'Editors' : 'Editor');
+    const editorsValue = page.querySelector('.editors');
+    editorsValue.innerHTML = html;
+
+    const editorsGroup = page.querySelector('.editorsGroup');
+    if (editors.length) {
+        editorsGroup.classList.remove('hide');
+    } else {
+        editorsGroup.classList.add('hide');
+    }
```
Comment (dmitrylyzo) [inline src/controllers/itemDetails/index.js:1061]:
```suggestion
    editorsGroup.classList.toggle('hide', !editors.length);
```

File: src/controllers/itemDetails/index.js [hunk 3]
```diff
@@ -1002,6 +1003,93 @@ function renderStudio(page, item, context) {
...
+    translatorsLabel.innerHTML = globalize.translate(translators.length > 1 ? 'Translators' : 'Translator');
+    const translatorsValue = page.querySelector('.translators');
+    translatorsValue.innerHTML = html;
+
+    const translatorsGroup = page.querySelector('.translatorsGroup');
+    if (translators.length) {
+        translatorsGroup.classList.remove('hide');
+    } else {
+        translatorsGroup.classList.add('hide');
+    }
```
Comment (dmitrylyzo) [inline src/controllers/itemDetails/index.js:1090]:
```suggestion
    translatorsGroup.classList.toggle('hide', !translators.length);
```

File: src/strings/en-us.json [hunk 4]
```diff
@@ -1728,5 +1728,20 @@
...
+    "Penciller": "Penciller",
+    "Inker": "Inker",
+    "Colorist": "Colorist",
+    "Letterer": "Letterer",
+    "CoverArtist": "Cover artist",
+    "Editors": "Editors",
+    "Editor": "Editor",
+    "Translators": "Translators",
+    "Translator": "Translator",
+    "HeaderCreators": "Creators"
```
Comment (dmitrylyzo) [inline src/strings/en-us.json:1746]: It would be better to put them in alphabetical order to minimize potential conflicts.

Comment (Pithaya) [reply src/strings/en-us.json:1746]: Should I sort the entire json file, or juste those new values ?

Comment (dmitrylyzo) [reply src/strings/en-us.json:1746]: Just new values.

File: src/strings/en-us.json [hunk 5]
```diff
@@ -1728,5 +1728,20 @@
...
+    "Creators": "Creators",
+    "Editor": "Editor",
+    "Editors": "Editors",
+    "HeaderCreators": "Creators",
+    "Illustrator": "Illustrator",
+    "Inker": "Inker",
+    "Letterer": "Letterer",
+    "Penciller": "Penciller",
+    "Translator": "Translator",
+    "Translators": "Translators"
```
Comment (dmitrylyzo) [inline src/strings/en-us.json:1746]: I meant, put the new values to where they should be in alphabetical order in the whole file.
I believe the first part of the file is ordered correctly.

**Comment by cvium** [comment]
@jellyfin/web server PR has been merged

**Comment by thornbill** [comment]
I pulled out adding the additional people types and allowing people to be edited for books into #5459 and #4361 fixed only actors being displayed on the item details page so I think this can be closed. Thanks for your contribution!

**Pull request closed without merging**